spring-data-jpa/src/test/java/org/springframework/data/jpa/domain/sample/ItemId.java

```diff
@@ -56,14 +58,12 @@ public void setManufacturerId(Integer manufacturerId) {
 	public boolean equals(Object o) {
 		if (this == o)
 			return true;
-		if (!(o instanceof ItemId))
+		if (!(o instanceof ItemId itemId))
 			return false;
 
-		ItemId itemId = (ItemId) o;
-
-		if (id != null ? !id.equals(itemId.id) : itemId.id != null)
+		if (!Objects.equals(id, itemId.id))
 			return false;
-		return manufacturerId != null ? manufacturerId.equals(itemId.manufacturerId) : itemId.manufacturerId == null;
+		return Objects.equals(manufacturerId, itemId.manufacturerId);
 	}
```

Comment:
We have a new template for `equals` at https://github.com/spring-projects/spring-data-build/blob/main/etc/ide/IntelliJ.equals-template.txt, which I'll run this by, to ensure we are compliant.